src/relayedTransactionV2Builder.ts

```diff
@@ -0,0 +1,59 @@
+import {Transaction} from "./transaction";
+import {TransactionPayload} from "./transactionPayload";
+import {AddressValue, BytesValue, ContractFunction, U64Value} from "./smartcontracts";
+import {IAddress, IChainID, IGasLimit, INonce, ITransactionPayload} from "./interface";
+import {ISigner} from "@elrondnetwork/erdjs-walletcore/out/interface";
+import {INetworkConfig} from "./interfaceOfNetwork";
+
+export class RelayedTransactionV2Builder {
+
+    /**
+     * Generate a relay v2 transaction to pay the gas for our users.
+     *
+     * @param {ISigner} innerTransactionSenderAccount The signer of the inner transaction
+     * @param {IAddress} innerTransactionReceiver The recipient address of the inner transaction
+     * @param {INonce} innerTransactionNonce The nonce of the inner transaction
+     * @param {ITransactionPayload} innerTransactionData The data field of the inner transaction
+     * @param {IGasLimit} innerTransactionGasLimit The gas limit of the inner transaction. It will be used for the big transaction
+     * @param {IChainID} chainID The Chain ID of the inner transaction
+     * @param {INetworkConfig} networkConfig The network config to be used for computing the gas units
+     * @returns {Transaction} Transaction containing the relayed transaction
+     */
+    public async buildRelayedTransactionV2(
+        innerTransactionSenderAccount: ISigner,
```

Comment:
Would it be possible to allow the caller provide the inner transaction as an argument, instead of these fields as arguments?

Reply:
:+1:

Reply:
Right, allow the caller to have both options

Reply:
sent the inner tx, which has to be already signed and with gas limit not 0